aws/ec2.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultTimeOut = 20
+	defaultRetries = 3
+)
+
+func convertError(err error) string {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return fmt.Sprintf("%s: %s", awsErr.Code(), awsErr.Message())
+	}
+
+	return fmt.Sprintf("%v", err.Error())
+}
+
+func retriable(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		switch awsErr.Code() {
+		case "InternalError":
+			return true
+		case "InternalFailure":
+			return true
+		case "RequestLimitExceeded":
+			return true
+		case "ServiceUnavailable":
+			return true
+		case "Unavailable":
+			return true
+		}
+	}
+
+	return false
+}
+
+func checkSourceDestinationValueIsDisable(check string) bool {
+	return check == "disable"
```

Comment:
nit: we could assign these strings to a constant so that if they change at any point, we'll just need to change the constant. Since the `disable` value is only really passed in for the tests, this isn't really necessary at this point.

Reply:
Yep. We already have constants for these in the libcalico-go API, so we should be using those throughout IIUC.